dev/com.ibm.ws.org.apache.cxf.cxf.rt.databinding.jaxb.3.2/src/org/apache/cxf/jaxb/io/DataReaderImpl.java

```diff
@@ -46,11 +46,12 @@
 import org.apache.cxf.service.model.MessagePartInfo;
 
 public class DataReaderImpl<T> extends JAXBDataBase implements DataReader<T> {
-    private static final Logger LOG = LogUtils.getLogger(JAXBDataBinding.class);
+    private static final Logger LOG = LogUtils.getLogger(DataReaderImpl.class);  // Liberty change
     JAXBDataBinding databinding;
     boolean unwrapJAXBElement;
     ValidationEventHandler veventHandler;
     boolean setEventHandler = true;
+    private static boolean isLoggableFine = LOG.isLoggable(Level.FINE);  // Liberty change
```

Comment:
Should this be `final`?

Reply:
On further thought, I don't think we should cache this check at the class level. If we do, we can't dynamically update the trace without restarting the server/application.

You could cache it at the start of a trace heavy method if performance is concerned.